rpc_util.go

```diff
@@ -295,6 +296,27 @@ func (o FailFastCallOption) before(c *callInfo) error {
 }
 func (o FailFastCallOption) after(c *callInfo, attempt *csAttempt) {}
 
+// OnFinish returns a CallOption that configures a callback to be called when
+// the call completes.
+func OnFinish(onFinish func(err error)) CallOption {
+	return OnFinishCallOption{
+		onFinish: onFinish,
+	}
+}
+
+// OnFinishCallOption is CallOption that indicates a callback to be called when
+// the call completes.
+type OnFinishCallOption struct {
+	onFinish func(error)
+}
+
+func (o OnFinishCallOption) before(c *callInfo) error {
+	c.onFinish = o.onFinish
```

Comment:
Pinged you about this offline; will leave as is for now pending further discussion.

Reply:
Ok, did this; c.onFinish = append(c.onFinish, o.OnFinish). I'm assuming we can get concurrency problems here to so did it this way? Is this wrong? Should I just append to the created memory rather than copy?

Reply:
Sorry, actually this doesn't make any sense wrt to not just appending to it. The other one below was call options into a local var. This one still writes to it. Plus, this is a safe write since it doesn't get concurrently read like the struct of calloptions below?